lib/datadog/open_feature/exposures/worker.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require_relative '../../core/workers/queue'
+require_relative '../../core/workers/polling'
+
+require_relative 'buffer'
+require_relative 'batch_builder'
+
+module Datadog
+  module OpenFeature
+    module Exposures
+      # This class is responsible for sending exposures to the Agent
+      class Worker
+        include Datadog::Core::Workers::Queue
+        include Datadog::Core::Workers::Polling
+
+        DEFAULT_FLUSH_INTERVAL_SECONDS = 30
+        DEFAULT_BUFFER_LIMIT = Buffer::DEFAULT_LIMIT
+
+        attr_reader :logger
+
+        def initialize(
+          settings:,
+          transport:,
+          logger: Datadog.logger,
+          flush_interval_seconds: DEFAULT_FLUSH_INTERVAL_SECONDS,
+          buffer_limit: DEFAULT_BUFFER_LIMIT
+        )
+          @logger = logger
+          @transport = transport
+          @batch_builder = BatchBuilder.new(settings)
+          @buffer_limit = buffer_limit
+          @flush_mutex = Mutex.new
+
+          self.buffer = Buffer.new(buffer_limit)
+          self.loop_base_interval = flush_interval_seconds
+          self.enabled = true
+        end
+
+        def start
+          return if !enabled? || running? || forked?
+
+          perform
+        end
+
+        def stop(force_stop = false, timeout = Core::Workers::Polling::DEFAULT_SHUTDOWN_TIMEOUT)
+          result = super
+          flush
+
+          result
+        end
+
+        def enqueue(event)
+          return false if forked?
+
+          buffer.push(event)
+
+          flush if buffer.length >= @buffer_limit
+          start unless running?
+
+          true
+        end
+
+        def dequeue
+          buffer.pop
+        end
+
+        def flush
+          events, dropped = dequeue
+          send_events(events || [], dropped.to_i)
+        end
+
+        def perform(*args)
+          events, dropped = args
+          send_events(events || [], dropped.to_i)
+        end
+
+        private
+
+        def send_events(events, dropped)
+          return if events.empty?
+
+          if dropped.positive?
+            logger.debug { "OpenFeature: Exposure worker dropped #{dropped} event(s) due to full buffer" }
+          end
+
+          payload = @batch_builder.payload_for(events)
+          send_payload(payload)
+        end
+
+        def send_payload(payload)
+          @flush_mutex.synchronize do
+            response = @transport.send_exposures(payload)
+            logger.debug { "OpenFeature: Send exposures response was not OK: #{response.inspect}" } unless response&.ok?
```

Comment:
This log is visible for our customers - is "exposures" a term that we are using a lot in our public documentation and UI?